heka/sandbox/outputs/redshift.lua

```diff
@@ -74,19 +74,75 @@ flush_interval = flush_interval * 1e9
+    {"appUpdateChannel",           "Fields[appUpdateChannel]",           "VARCHAR",   30},
+    {"appBuildId",                 "Fields[appBuildId]",                 "VARCHAR",   30},
+    {"appVendor",                  "Fields[appVendor]",                  "VARCHAR",   30},
+    {"clientId",                   "Fields[clientId]",                   "VARCHAR",   100},
+    {"sampleId",                   "Fields[sampleId]",                   "SMALLINT",  nil},
+    {"environment_addons",         "Fields[environment.addons]",         "VARCHAR",   MAX_LENGTH},
+    {"environment_build",          "Fields[environment.build]",          "VARCHAR",   MAX_LENGTH},
+    {"environment_partner",        "Fields[environment.partner]",        "VARCHAR",   MAX_LENGTH},
+    {"environment_profile",        "Fields[environment.profile]",        "VARCHAR",   MAX_LENGTH},
+    {"environment_settings",       "Fields[environment.settings]",       "VARCHAR",   MAX_LENGTH},
+    {"environment_system",         "Fields[environment.system]",         "VARCHAR",   MAX_LENGTH},
+    {"payload",                    "Payload",                            "VARCHAR",   MAX_LENGTH},
+    {"payload_addonDetails",       "Fields[payload.addonDetails]",       "VARCHAR",   MAX_LENGTH},
+    {"payload_addonHistograms",    "Fields[payload.addonHistograms]",    "VARCHAR",   MAX_LENGTH},
+    {"payload_childPayloads",      "Fields[payload.childPayloads]",      "VARCHAR",   MAX_LENGTH},
+    {"payload_chromeHangs",        "Fields[payload.chromeHangs]",        "VARCHAR",   MAX_LENGTH},
+    {"payload_fileIOReports",      "Fields[payload.fileIOReports]",      "VARCHAR",   MAX_LENGTH},
+    {"payload_histograms",         "Fields[payload.histograms]",         "VARCHAR",   MAX_LENGTH},
+    {"payload_info",               "Fields[payload.info]",               "VARCHAR",   MAX_LENGTH},
+    {"payload_keyedHistograms",    "Fields[payload.keyedHistograms]",    "VARCHAR",   MAX_LENGTH},
+    {"payload_lateWrites",         "Fields[payload.lateWrites]",         "VARCHAR",   MAX_LENGTH},
+    {"payload_log",                "Fields[payload.log]",                "VARCHAR",   MAX_LENGTH},
+    {"payload_simpleMeasurements", "Fields[payload.simpleMeasurements]", "VARCHAR",   MAX_LENGTH},
+    {"payload_slowSQL",            "Fields[payload.slowSQL]",            "VARCHAR",   MAX_LENGTH},
+    {"payload_slowSQLstartup",     "Fields[payload.slowSQLstartup]",     "VARCHAR",   MAX_LENGTH},
+    {"payload_threadHangStats",    "Fields[payload.threadHangStats]",    "VARCHAR",   MAX_LENGTH},
+    {"payload_UIMeasurements",     "Fields[payload.UIMeasurements]",     "VARCHAR",   MAX_LENGTH},
+    {"http_DNT",                   "Fields[DNT]",                        "BOOLEAN",   nil},
+    {"http_Date",                  "Fields[Date]",                       "TIMESTAMP", nil}
+}
+
+function make_create_table()
+    local sql = "CREATE TABLE IF NOT EXISTS " .. table_name .. " ("
```

Comment:
In Lua you really don't want to build up strings with `..` as each concatenation creates a new immutable string.